Packs/DigitalShadows/Integrations/integration-DigitalShadows.yml

```diff
@@ -750,19 +750,30 @@ script:
 
 
     var fetchCommandIncidents = function(comnd, bodyArgs, name){
-        incidents = [];
         cmd = commandDictionary[comnd];
+        bodyArgs.pagination_size = 500;
+        bodyArgs.pagination_offset = 0;
+        return fetchCommandIncidentPages(cmd, bodyArgs, name);
+    };
+
+    var fetchCommandIncidentPages = function(cmd, bodyArgs, name) {
+        incidents = [];
         res = sendRequest(url+cmd.url, cmd.method, createBody(bodyArgs, arrayArgs));
         parsed_res = JSON.parse(res);
         if (parsed_res && parsed_res.content){
             parsed_res.content.forEach(function(inc){
                 inc['origin'] = name;
                 incidents.push({
-                  name: name,
-                  rawJSON: JSON.stringify(inc)
+                    name: name,
+                    rawJSON: JSON.stringify(inc)
                 });
             });
         }
+        if ( parsed_res.total > (bodyArgs.pagination_size + bodyArgs.pagination_offset)) {
+            bodyArgs.pagination_offset = bodyArgs.pagination_offset + bodyArgs.pagination_size ;
+            incidents = incidents.concat(fetchCommandIncidentPages(cmd, bodyArgs, name));
+        }
```

Comment:
I believe we will need to make some changes to the logic here.
Currently, we are adding pagination without adding any fetch limit, which may cause problems in case the number of incidents is more than what XSOAR can handle (for example, 200-300 incidents in total).

To address this issue, I suggest the following course of action:
1. Adding a `fetch_limit` parameter to the integration, as can be seen in many other integrations.
2. Using the integration context to save all relevant fetch variables, so in case we have over 100 incidents (for example, this limit is user-configurable), we can return 100 and continue where we left off on the next fetch cycle

Moreover, I suggest avoiding recursion and using an iterative approach for integration commands.

Feel free to reach out if you have any questions or need my advice with doing this fix - it isn't an easy one, but I'm happy to offer my advice and assistance.

Reply:
Done as per our conversation.